### ISSUE TITLE
support Multi-Module workspace upgrades

### ISSUE THREAD

**Issue by xmlking** [oligot/go-mod-upgrade#25]
` go-mod-upgrade` not supporting go1.18 workspace enabled projects. 
$ go-mod-upgrade            
                                                                                                                                                                                  
   ⨯ upgrade failed            error=Error running go command to discover modules: exit status 1 stderr=go: -mod may only be set to readonly when in workspace mode, but it is set to "mod"
        Remove the -mod flag to use the default readonly value,
        or set GOWORK=off to disable workspace mode.

I have Multi-Module workspace repo. 

```
$ go list -json -m                                                                                                                                                                                             
{
        "Path": "github.com/tz/schannel",
        "Main": true,
        "Dir": "/Users/schintha/Developer/Work/go/schannel",
        "GoMod": "/Users/schintha/Developer/Work/go/schannel/go.mod",
        "GoVersion": "1.18"
}
{
        "Path": "github.com/tz/schannel/apps/lockboard",
        "Main": true,
        "Dir": "/Users/schintha/Developer/Work/go/schannel/apps/lockboard",
        "GoMod": "/Users/schintha/Developer/Work/go/schannel/apps/lockboard/go.mod",
        "GoVersion": "1.18"
}
{
        "Path": "github.com/tz/schannel/cmd/agent",
        "Main": true,
        "Dir": "/Users/schintha/Developer/Work/go/schannel/cmd/agent",
        "GoMod": "/Users/schintha/Developer/Work/go/schannel/cmd/agent/go.mod",
        "GoVersion": "1.18"
}
{
        "Path": "github.com/tz/schannel/service/engine",
        "Main": true,
        "Dir": "/Users/schintha/Developer/Work/go/schannel/service/engine",
        "GoMod": "/Users/schintha/Developer/Work/go/schannel/service/engine/go.mod",
        "GoVersion": "1.18"
}

```
`GOWORK=off go-mod-upgrade` can only update root module. 

looking forward for `Multi-Module` upgrade feature.

**Comment by oligot** [comment]
Hi,

I just pushed a commit on the main branch to support the Go 1.18 workspace.
Could you try it on your project and see if it works now ?

**Comment by xmlking** [comment]
I am getting error 
<img width="1563" alt="image" src="https://user-images.githubusercontent.com/627951/161361073-3af800c0-39a6-4185-b28b-bdbf90a9dcf6.png">

**Comment by xmlking** [comment]
when I supply `GOWORK=off` 

<img width="793" alt="image" src="https://user-images.githubusercontent.com/627951/161361133-373092c1-d473-4abf-ab43-b582c62286fb.png">

**Comment by oligot** [comment]
Hi,

Could you share a small repo or at least your go.mod/go.work files so that we can reproduce the bug on our side ?

**Comment by xmlking** [comment]
https://github.com/xmlking/toolkit

<img width="1539" alt="image" src="https://user-images.githubusercontent.com/627951/161485952-9a1d6f69-0621-4488-8bbc-200f3a09256b.png">

I have another private repo that has `go.work` as  
```
go 1.18

use (
	./
	./apps/lockboard
	./cmd/agent
	./cmd/tz
	./service/engine
)

replace github.com/xmlking/toolkit/xds => github.com/xmlking/toolkit/xds v0.2.5-0.20220329222226-895ec254c86c

replace github.com/xmlking/toolkit => github.com/xmlking/toolkit v0.2.5-0.20220329222226-895ec254c86c

```

**Comment by xmlking** [comment]
for my https://github.com/xmlking/toolkit if I use `GOWORK=off go-mod-upgrade` 
<img width="982" alt="image" src="https://user-images.githubusercontent.com/627951/161486506-462af496-6096-4002-845f-7a255f8e4e83.png">

**Comment by oligot** [comment]
I just pushed a new commit on the main brain.
Could you try again ?

**Comment by xmlking** [comment]
works now. Thank you for contributing this great tool and  quick support
<img width="736" alt="image" src="https://user-images.githubusercontent.com/627951/161561608-c5e01404-5322-471b-8b64-53e878287279.png">

**Comment by oligot** [comment]
Thanks for your quick feedback :pray:
Closing this issue as this is now resolved.

**Comment by xmlking** [comment]
can you cut a new release ?

**Comment by oligot** [comment]
Just released [version v0.9.0](https://github.com/oligot/go-mod-upgrade/releases/tag/v0.9.0)
